latch_cli/centromere/utils.py

```diff
@@ -214,32 +229,28 @@ def __exit__(self, *args):
         self.cleanup(*args)
 
     def create(self, *args):
-        if not self.remote:
+        if self.ssh_client is None:
             self._tempdir = tempfile.TemporaryDirectory()
             return Path(self._tempdir.name).resolve()
 
-        if self.internal_ip is None or self.username is None:
-            raise ValueError("Must provide an ssh connection if remote is True.")
-
-        td = "".join(
-            random.choices(
-                string.ascii_uppercase + string.ascii_lowercase + string.digits, k=8
-            )
-        )
+        td = build_random_dir_name()
 
         self._tempdir = f"/tmp/{td}"
 
-        client = _construct_ssh_client(self.internal_ip, self.username)
-        client.exec_command(f"mkdir {self._tempdir}")
+        self.ssh_client.exec_command(f"mkdir {self._tempdir}")
         return self._tempdir
 
     def cleanup(self, *args):
-        if (
-            not self.remote
-            and self._tempdir is not None
-            and not isinstance(self._tempdir, str)
-        ):
+        if self.ssh_client is not None:
+            self.ssh_client.exec_command(f"rm -rf {self._tempdir}")
+            return
+        if self._tempdir is not None and not isinstance(self._tempdir, str):
             self._tempdir.cleanup()
-        elif not (self.internal_ip is None or self.username is None):
-            client = _construct_ssh_client(self.internal_ip, self.username)
-            client.exec_command(f"rm -rf {self._tempdir}")
+
+
+def build_random_dir_name() -> str:
```

Comment:
```suggestion
def get_random_string(k: int) -> str:
```

rename + pass `k` in as a param